common/network-common/src/main/java/org/apache/spark/network/server/TransportServer.java

```diff
@@ -105,11 +104,13 @@ private void init(String hostToBind, int portToBind) {
     EventLoopGroup workerGroup =  NettyUtils.createEventLoop(ioMode, conf.serverThreads(),
       conf.getModuleName() + "-server");
 
+    String name = System.getProperty("os.name");
+    boolean isNotWindows = 7 > name.length() || !name.regionMatches(true, 0, "Windows", 0, 7);
```

Comment:
We dont need the `7 > name.length() ||` here

Reply:
It's originated from the previous `commons-lang3` code. Initially, I also thought like you. However, after Liang-Chi commented, I realized that we need it in the following case.

If `System.getProperty("os.name")` == `Win`, `7 > name.length()` will return true here. It means `isNotWindows = true`.  Otherwise, `isNotWindows` becomes `false` for `Win`. 
```
jshell> String name = "Win"
name ==> "Win"

jshell> !name.regionMatches(true, 0, "Windows", 0, 1)
$3 ==> false
```

Reply:
```
scala> val name = "Win"
name: String = Win

scala> !name.regionMatches(true, 0, "Windows", 0, 7)
res2: Boolean = true

scala> 
```

It should be `0, 7)`, no ?

Reply:
Hmm, the length check comes from commons-lang3's `startsWith`.

I recommended to also add the check as I thought that `regionMatches` will throw `StringIndexOutOfBoundsException` if the length of `str` is less than preLen`, based on search answer from Google.

But seems in @mridulm's test above, it doesn't. The source code of `regionMatches` can confirm it.


```java
public boolean regionMatches(int toffset, String other, int ooffset, int len) {
        // Note: toffset, ooffset, or len might be near -1>>>1.
        if ((ooffset < 0) || (toffset < 0) ||
             (toffset > (long)length() - len) ||
             (ooffset > (long)other.length() - len)) {
            return false;
        }
...
}

Reply:
Oh, got it. It seems that I made a mistake yesterday during testing.
> It should be 0, 7), no ?

Thank you, @mridulm and @viirya . Let me do the follow-up.

Reply:
Here is a follow-up.
- https://github.com/apache/spark/pull/51524